tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppFilterImpl.scala

```diff
@@ -71,21 +73,37 @@ class AppFilterImpl(
 
     val filterAppName = appArgs.applicationName.getOrElse("")
     if (appArgs.applicationName.isSupplied && filterAppName.nonEmpty) {
-      val filtered = apps.filter { app =>
-        val appNameOpt = app.appInfo.map(_.appName)
-        if (appNameOpt.isDefined) {
-          appNameOpt.get.equals(filterAppName)
-        } else {
-          // in complete log file
-          false
-        }
+      val filtered = if (filterAppName(0).equals(negate)) {
+        apps.filterNot(app => filterAppsNegate(app, filterAppName))
+      } else {
+        apps.filter(app => filterApps(app, filterAppName))
       }
       filtered.map(_.eventlog).toSeq
     } else {
       apps.map(x => x.eventlog).toSeq
     }
   }
 
+  def filterApps(app: AppFilterReturnParameters, filterAppName: String): Boolean = {
+    val appNameOpt = app.appInfo.map(_.appName)
+    if (appNameOpt.isDefined) {
+      appNameOpt.get.contains(filterAppName)
+    } else {
+      // in complete log file
+      false
+    }
+  }
+
+  def filterAppsNegate(app: AppFilterReturnParameters, filterAppName: String): Boolean = {
```

Comment:
doesn't seem like we need function for this, just remove the ~ before passing it into the filterApps function